Cannot find module 'gpt-3-encoder' or its corresponding type declarations.

rossanodr opened the pull request:
Hey man. I'm just a nobbie but I'm trying to learn with your project.
After I run npm install @syonfox/gpt-3-encoder
I can't import your project.
import {encode, decode, countTokens, tokenStats} from "gpt-3-encoder"
I am getting the error 'Cannot find module 'gpt-3-encoder' or its corresponding type declarations."
:(
I'm using next js

syonfox committed:
Add count tokens function

I don't actually want to encode into tokens for my use case, quickly count to check my request won't exceed the limit.

This should be faster since we don't initialize the memory for the output array.

```
const crypto = require('crypto');
// Generate a random string of a given length
function generateRandomString(length) {
    return crypto.randomBytes(length).toString('hex');
}

const {encode, decode, countTokens} = require('gpt-3-encoder')

let str = 'This is an example sentence to try encoding out on!'
// let now = Date.now();
let encoded = encode(str)
console.log('Encoded this string looks like: ', encoded)
console.log('We can look at each token and what it represents)
let tokencount = 0;
for(let token of encoded){
    tokencount ++;
    console.log({token, string: decode([token])})
}
console.log("there are n tokens: ", tokencount);
let decoded = decode(encoded)
console.log('We can decode it back into:\n', decoded)

let now = Date.now();
// todo: write an benchmark for the above method vs  int countTokens(str)
str = generateRandomString(10000);

console.time('fencode');
encoded = encode(str);
console.log(`First encode to cache string n stuff in mem`);
console.timeEnd('fencode');


console.log(`Original string length: ${str.length}`);
// Benchmark the encode function
console.time('encode');
encoded = encode(str);
console.log(`Encoded string length: ${encoded.length}`);
console.timeEnd('encode');

// Benchmark the countTokens function
console.time('countTokens');
let tokenCount = countTokens(str);
console.log(`Number of tokens: ${tokenCount}`);
console.timeEnd('countTokens');


console.log(`Original string length: ${str.length}`);
console.log(`Encoded string length: ${encoded.length}`);
console.log(`Number of tokens: ${tokenCount}`);
```

```
We can decode it back into:
 This is an example sentence to try encoding out on!
First encode to cache string n stuff in mem
fencode: 163.57ms
Original string length: 20000
Encoded string length: 11993
encode: 124.265ms
Number of tokens: 11993
countTokens: 29.2ms
Original string length: 20000
Encoded string length: 11993
Number of tokens: 11993

```

syonfox committed:
add algo doc

NickHeiner committed:
Fix bug with BPE cache (#1)

Co-authored-by: Andrew Healey <doolie.healey@gmail.com>

NickHeiner committed:
Add count tokens function (#2)

I don't actually want to encode into tokens for my use case, quickly count to check my request won't exceed the limit.

This should be faster since we don't initialize the memory for the output array.

```
const crypto = require('crypto');
// Generate a random string of a given length
function generateRandomString(length) {
    return crypto.randomBytes(length).toString('hex');
}

const {encode, decode, countTokens} = require('gpt-3-encoder')

let str = 'This is an example sentence to try encoding out on!'
// let now = Date.now();
let encoded = encode(str)
console.log('Encoded this string looks like: ', encoded)
console.log('We can look at each token and what it represents)
let tokencount = 0;
for(let token of encoded){
    tokencount ++;
    console.log({token, string: decode([token])})
}
console.log("there are n tokens: ", tokencount);
let decoded = decode(encoded)
console.log('We can decode it back into:\n', decoded)

let now = Date.now();
// todo: write an benchmark for the above method vs  int countTokens(str)
str = generateRandomString(10000);

console.time('fencode');
encoded = encode(str);
console.log(`First encode to cache string n stuff in mem`);
console.timeEnd('fencode');


console.log(`Original string length: ${str.length}`);
// Benchmark the encode function
console.time('encode');
encoded = encode(str);
console.log(`Encoded string length: ${encoded.length}`);
console.timeEnd('encode');

// Benchmark the countTokens function
console.time('countTokens');
let tokenCount = countTokens(str);
console.log(`Number of tokens: ${tokenCount}`);
console.timeEnd('countTokens');


console.log(`Original string length: ${str.length}`);
console.log(`Encoded string length: ${encoded.length}`);
console.log(`Number of tokens: ${tokenCount}`);
```

```
We can decode it back into:
 This is an example sentence to try encoding out on!
First encode to cache string n stuff in mem
fencode: 163.57ms
Original string length: 20000
Encoded string length: 11993
encode: 124.265ms
Number of tokens: 11993
countTokens: 29.2ms
Original string length: 20000
Encoded string length: 11993
Number of tokens: 11993

```

Co-authored-by: Kier <syonfox@users.noreply.github.com>

Commit:
Add fork notice

Commit:
Bump version

syonfox committed:
Update index.d.ts

syonfox committed:
Update index.js

Commit:
Merge branch 'master' into syonfox-dev

Commit:
Merge remote-tracking branch 'origin/patch-1' into syonfox-dev

# Conflicts:
#	Encoder.js

syonfox committed:
Merge pull request #1 from NickHeiner/master

merge back the one other change and follow nick with moving to version 1.2 for feature add.

Commit:
Merge remote-tracking branch 'origin/patch-1' into syonfox-dev

Commit:
Add tokenStats some tests, and some error checks, also docs and readme updates

maybe we should host jsdoc on github pages

also performed npm audit and updates for security vulnerabilities.

Commit:
add docs to repo for everyone else

Commit:
update readme and docs with links to pages

Commit:
update docs and bump

Commit:
update docs and bump

Commit:
add a test for issue https://github.com/latitudegames/GPT-3-Encoder/issues/9

It seems to work fine

Commit:
clean up test a bit

Commit:
fix #15 in another way by removing textEncoder dependency

By using the Buffer class in this way, you can encode and decode strings as UTF-8 without using the TextEncoder and TextDecoder classes. This can be useful if you need to support earlier versions of Node.js that do not have these classes in the util module.

Commit:
fix #15 bump semi major version to 1.3.0 for possibly breaking change but i think its actually mor widely compatible and should be fine.

Commit:
update readme

Commit:
bump n docs

Commit:
Som more docs and also refactor so that we can use it in the browser by webpaking... todo need to test browser version
This precomputes bpe_ranks so that it is it the from needed make sure to rebuild if you change the bpe dump file

Commit:
bump semi major version for browser support and loading change

Commit:
added browser suport with browserify and an example

Commit:
bump

Commit:
readme and fix npm rc

Commit:
update docs ok lets test it noe

Commit:
hope this pushes demo to github pages

Commit:
hope this pushes demo to github pages

Commit:
hope this pushes demo to github pages

Commit:
Some fixes to all of this hope its stable now for 1.4 release

Commit:
Some fixes to all of this hope its stable now for 1.4 release

Commit:
revert npm for semi major release 1.2.0 latitudegames ... note you will need to deploy docs to github pages for demo to work

syonfox committed:
Merge branch 'master' into latpull

syonfox commented on package.json:
```diff
@@ -1,29 +1,55 @@
-  "name": "gpt-3-encoder",
-  "version": "1.1.3",
-  "description": "Javascript BPE Encoder Decoder for GPT-2 / GPT-3",
+  "name": "@syonfox/gpt-3-encoder",
```
#30 @rossanodr  

I believe when you npm install the "package manager" uses the name to specify the package from here.  

SO if you install the @syonfox version you need to also used @syonfox/gpt-3-encoder

hashfox committed:
https://github.com/latitudegames/GPT-3-Encoder/pull/30#issuecomment-1460069823

Add types let me know if this is the correct way to implement it.

And here are some additional useful stats that could be added to the stats object:

    averageWordLength: The average length of the words in the tokens array.
    mostFrequentWords: An array of the most frequently occurring words in the tokens array, ordered by frequency.
    leastFrequentWords: An array of the least frequently occurring words in the tokens array, ordered by frequency.
    wordPositions: An object that maps each word in the tokens array to its positions (indices) in the array.
    wordLengths: An array of the lengths of the words in the tokens array.

hashfox committed:
Merge branch 'master' into latpull

# Conflicts:
#	docs/Encoder.js.html
#	docs/global.html
#	docs/index.html
#	package-lock.json

hashfox committed:
Fix token stats to run, and bump npm

hashfox committed:
Rebuild docs

syonfox committed:
note

syonfox committed:
new rc relaease

syonfox committed:
new rc relaease

syonfox committed:
add gptoken runner

syonfox committed:
Update README.md
